sdk/src/main/java/com/mongodb/stitch/android/services/mongodb/MongoClient.java

```diff
@@ -247,54 +110,78 @@ public PipelineStage makeDeleteStage(
          * of the request.
          */
         public Task<List<Document>> find(final Document query, final Document projection, final Integer limit) {
-            return convertToDocuments(_database._client._stitchClient.executePipeline(makeFindStage(query, projection, limit, null)));
+            Document doc = new Document(Parameters.QUERY, query);
+            doc.put(Parameters.DATABASE, _database._dbName);
+            doc.put(Parameters.COLLECTION, _collName);
+            doc.put(Parameters.LIMIT, limit);
+
+            if (projection != null) {
+                doc.put(Parameters.PROJECT, projection);
+            }
+
+            return _database._client._stitchClient.executeServiceFunction(
+                "find", _database._client._service, doc
+            ).continueWith(new Continuation<Object, List<Document>>() {
+                @Override
+                public List<Document> then(@NonNull Task<Object> task) throws Exception {
+                    if (task.isSuccessful()) {
+                        Object result = task.getResult();
+                        if (result instanceof JSONArray) {
+                            List<Document> docs = new ArrayList<>();
+                            for (int i = 0; i < ((JSONArray) result).length(); i++) {
```

Comment:
Could probably use fancy for loop here

Reply:
foreach is not applicable with `JSONArray`